tests/test_run.py

```diff
@@ -44,15 +45,85 @@ def test_default_sends_verify_true():
+        mock_adapter_send.return_value = resp
         httsleep.run()
-        assert mock_session_send.called
-        args, kwargs = mock_session_send.call_args
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
     assert 'verify' in kwargs
     assert kwargs['verify'] == True
 
 
+@httpretty.activate
+def test_default_uses_default_session():
+    """Should use a default Session object unless one is specified"""
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200})
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == requests.utils.default_headers()
+    assert 'verify' in kwargs
+    assert kwargs['verify'] == True
+
+
+@httpretty.activate
+def test_propagate_session():
+    """Should propagate a Session's headers, verify setting, cookies etc. when specified"""
+    session = requests.Session()
+    session.cookies = {'tasty-cookie': 'chocolate'}
+    session.headers = {'Content-Type': 'test/type'}
+    session.verify = '/session/verify'
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200}, session=session)
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == {'Content-Type': 'test/type', 'Cookie': 'tasty-cookie=chocolate'}
+    assert 'verify' in kwargs
+    assert kwargs['verify'] == '/session/verify'
+
+
+@httpretty.activate
+def test_session_headers_and_request_headers_combine():
+    """Should merge headers directly specified over any headers in a specified Session"""
+    session = requests.Session()
+    session.headers = {'session-header': 'mySession', 'conflict-header': 'session-loses'}
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200}, session=session,
+                          headers={'req-header': 'myRequest', 'conflict-header': 'req-wins'})
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == {'conflict-header': 'req-wins', 'req-header': 'myRequest', 'session-header': 'mySession'}
+
+
+@httpretty.activate
+def test_request_verify_overrules_session_verify():
+    """Should give precedence to the 'verify' setting in the request, over that in a specified Session"""
+    session = requests.Session()
+    session.verify = '/path/to/ca/bundle'
```

Comment:
Could we not set this to `True` and then do 

```
httsleep = HttSleeper(URL, {'status_code': 200}, session=session, verify=False)
```

And then check whether `verify` is `False`?

Reply:
That would of course work, but:
-  as per my other comment, `verify` can be a string as well
-  I thought it improved readability of the test to have the overriding value of `/override/path`.

I can change it to a boolean case if you'd prefer, just say the word.

Reply:
Yep, grand, leave it as it is